### PR TITLE
chore(lint): re-enable eslint plugin angular

### LINKED ISSUE #235: [Regression] ESLint plugin plugin:testing-library/angular & plugin:jest-dom/recommended are removed
@timdeschryver It seems like you (accidentally?) removed
```
"extends": ["plugin:testing-library/angular", "plugin:jest-dom/recommended"]
```
with the upgrade to/for Angular 12

[See Commit](https://github.com/testing-library/angular-testing-library/commit/e6d2e84470abdb6ff8c61a2494cc0fed2805ef72#diff-45673f3f4db2127124203aa2c44baf45de285dc6e30c793f368fd1829225a630)

both in `apps/example-app` and `apps/example-app-karma`

### PR DESCRIPTION
**Pull request by the-ult**
Re-enable the eslint testing-library plugins.
- and improve the eslint settings for other projects

fixes #235

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -102,20 +102,20 @@
...
       "files": ["*.html"],
       "rules": {}
     },
     {
       "files": ["*.ts", "*.js"],
       "extends": ["prettier"]
+    },
+    {
+      "files": ["*.spec.ts"],
+      "extends": ["plugin:testing-library/angular"],
```
Comment (the-ult) [inline .eslintrc.json:115]: `"plugin:testing-library/angular"` can be used by _all_ projects. Therefore added to the main eslint.json

File: apps/example-app/.eslintrc.json [hunk 2]
```diff
@@ -27,6 +27,17 @@
         ]
       }
     },
+    {
```
Comment (the-ult) [inline apps/example-app/.eslintrc.json:30]: jest specific plugins added on app/project level eslint.json

File: projects/testing-library/tests/render-template.spec.ts [hunk 3]
```diff
@@ -44,83 +44,86 @@ class GreetingComponent {
 }
 
 test('the directive renders', async () => {
-  const component = await render('<div onOff></div>', {
+  const view = await render('<div onOff></div>', {
     declarations: [OnOffDirective],
   });
 
-  expect(component.container.querySelector('[onoff]')).toBeInTheDocument();
+  // eslint-disable-next-line testing-library/no-container
```
Comment (the-ult) [inline projects/testing-library/tests/render-template.spec.ts:51]: Is there another/proper/better way to check if a `directive` is available in the `screen`?

Would something like: `screen.getByAttribute('[onoff]')` work?
Or maybe an Angular specific methode like `getByDirective()` ??

File: projects/testing-library/tests/find-by.spec.ts [hunk 4]
```diff
@@ -12,25 +12,25 @@ class FixtureComponent {
...
     await render(FixtureComponent);
     await expect(screen.findByText('I am invisible', {}, { timeout: 40 })).rejects.toThrow('x');
   });
 });
 
 describe('rendered component', () => {
-  test('waits for element to be added to the DOM', async () => {
-    const { findByText } = await render(FixtureComponent);
-    await expect(findByText('I am visible')).resolves.toBeTruthy();
+  it('waits for element to be added to the DOM', async () => {
```
Comment (timdeschryver) [inline projects/testing-library/tests/find-by.spec.ts:27]: These test should be using the utility methods returned from `render`.

File: projects/testing-library/tests/render-template.spec.ts (hunk 3 above)
Comment (timdeschryver) [reply projects/testing-library/tests/render-template.spec.ts:51]: Not that I know of (`getBytAttribute` doesn't exist, AFAIK), perhaps it makes sense to add a helper method that's specific to ATL.
I'm going to think about this.

File: projects/testing-library/tests/render.spec.ts [hunk 5]
```diff
@@ -22,15 +22,15 @@ import { render, fireEvent } from '../src/public_api';
 class FixtureComponent {}
 
 test('creates queries and events', async () => {
-  const component = await render(FixtureComponent);
+  await render(FixtureComponent);
```
Comment (timdeschryver) [inline projects/testing-library/tests/render.spec.ts:25]: This test should also be left, as-is.
It's here to test those methods, it's just considered a bad practice to use them.
Maybe we should try to deprecate them... but for now I would prefer to under test.

**Review by timdeschryver — COMMENTED** [review]
Hi, thanks for creating the PR and fixing the lint issues!
I noticed some tests are changed from `test` to `it`, was there a reason for this?
I would prefer to just use one syntax, if a linter prefers `it` that's good for me.

**Comment by the-ult** [comment]
> Hi, thanks for creating the PR and fixing the lint issues!
> I noticed some tests are changed from `test` to `it`, was there a reason for this?
> I would prefer to just use one syntax, if a linter prefers `it` that's good for me.

These where indeed 'auto' fixed by the linter.

When a test is standalone. Use `test()`
Within a `describe()` use `it()`

File: projects/testing-library/tests/render-template.spec.ts (hunk 3 above)
Comment (the-ult) [reply projects/testing-library/tests/render-template.spec.ts:51]: That would be really nice

**Comment by the-ult** [comment]
Hey @timdeschryver,

I resolved / fixed your comments.
Thanks for the review 👍

**Comment by timdeschryver** [comment]
Ah I didn't know that.
Thanks for the info 👍

File: projects/testing-library/tests/find-by.spec.ts [hunk 6]
```diff
@@ -12,25 +12,29 @@ class FixtureComponent {
...
     await render(FixtureComponent);
     await expect(screen.findByText('I am invisible', {}, { timeout: 40 })).rejects.toThrow('x');
   });
 });
 
 describe('rendered component', () => {
-  test('waits for element to be added to the DOM', async () => {
+  it('waits for element to be added to the DOM', async () => {
     const { findByText } = await render(FixtureComponent);
+    /// We whish to test the utility function from `render` here.
```
Comment (timdeschryver) [inline projects/testing-library/tests/find-by.spec.ts:29]:
```suggestion
    /// We wish to test the utility function from `render` here.
```

File: projects/testing-library/tests/find-by.spec.ts [hunk 7]
```diff
@@ -12,25 +12,29 @@ class FixtureComponent {
...
+    // eslint-disable-next-line testing-library/prefer-screen-queries
     await expect(findByText('I am visible')).resolves.toBeTruthy();
   });
 
-  test('rejects when something cannot be found', async () => {
+  it('rejects when something cannot be found', async () => {
     const { findByText } = await render(FixtureComponent);
+    /// We whish to test the utility function from `render` here.
```
Comment (timdeschryver) [inline projects/testing-library/tests/find-by.spec.ts:36]:
```suggestion
    /// We wish to test the utility function from `render` here.
```

File: projects/testing-library/tests/render.spec.ts [hunk 8]
```diff
@@ -22,15 +22,19 @@ import { render, fireEvent } from '../src/public_api';
...
 
 test('creates queries and events', async () => {
-  const component = await render(FixtureComponent);
-
-  fireEvent.input(component.getByTestId('input'), { target: { value: 'a super awesome input' } });
-  component.getByDisplayValue('a super awesome input');
-  fireEvent.click(component.getByText('button'));
+  const view = await render(FixtureComponent);
+
+  /// We whish to test the utility function from `render` here.
```
Comment (timdeschryver) [inline projects/testing-library/tests/render.spec.ts:27]:
```suggestion
  /// We wish to test the utility function from `render` here.
```

File: projects/testing-library/tests/render.spec.ts [hunk 9]
```diff
@@ -104,24 +108,29 @@ describe('Angular component life-cycle hooks', () => {
...
-  test('will call ngOnInit on initial render', async () => {
+  it('will call ngOnInit on initial render', async () => {
     const nameInitialized = jest.fn();
     const componentProperties = { nameInitialized };
-    const component = await render(FixtureWithNgOnChangesComponent, { componentProperties });
+    const view = await render(FixtureWithNgOnChangesComponent, { componentProperties });
 
-    component.getByText('Initial');
-    expect(nameInitialized).toBeCalledWith('Initial');
+    /// We whish to test the utility function from `render` here.
```
Comment (timdeschryver) [inline projects/testing-library/tests/render.spec.ts:116]:
```suggestion
    /// We wish to test the utility function from `render` here.
```

File: projects/testing-library/tests/render.spec.ts [hunk 10]
```diff
@@ -104,24 +108,29 @@ describe('Angular component life-cycle hooks', () => {
...
     const nameChanged = jest.fn();
     const componentProperties = { nameInitialized, nameChanged, name: 'Sarah' };
-    const component = await render(FixtureWithNgOnChangesComponent, { componentProperties });
 
-    component.getByText('Sarah');
-    expect(nameChanged).toBeCalledWith('Sarah', true);
-    // expect `nameChanged` to be called before `nameInitialized`
+    const view = await render(FixtureWithNgOnChangesComponent, { componentProperties });
+
+    /// We whish to test the utility function from `render` here.
```
Comment (timdeschryver) [inline projects/testing-library/tests/render.spec.ts:129]:
```suggestion
    /// We wish to test the utility function from `render` here.
```

**Review by timdeschryver — APPROVED** [review]

**Pull request merged**